src/pyfaf/actions/repodel.py

```diff
@@ -25,24 +27,40 @@ class RepoDel(Action):
 
 
     def run(self, cmdline, db):
-        repo = (db.session.query(Repo)
-                .filter(Repo.name == cmdline.REPO)
-                .first())
-
-        if not repo:
-            self.log_error("Repository '{0}' not found"
-                           .format(cmdline.REPO))
+        if not cmdline.REPO and not cmdline.all:
+            self.log_error("No repositories specified")
             return 1
 
-        for url in repo.url_list:
-            db.session.delete(url)
+        repos = []
+
+        if cmdline.all or '*' in cmdline.REPO:
+            repos.extend(db.session.query(Repo).all())
+        else:
+            for pattern in cmdline.REPO:
+                pattern = re.sub('_', r'\\_', pattern)
+                pattern = re.sub('%', r'\\%', pattern)
+                pattern = re.sub(r'\*', '%', pattern)
+                pattern = re.sub(r'\?', '_', pattern)
+                repos.extend(db.session.query(Repo).filter(Repo.name.like(pattern)).all())
+
+        if repos:
+            repos = sorted(list(set(repos)), key=lambda x: x.name)
+            for repo in repos:
+                for url in repo.url_list:
+                    db.session.delete(url)
 
-        self.log_info("Removing repository '{0}'".format(cmdline.REPO))
+                self.log_info("Removing repository '{0}'".format(repo))
 
-        db.session.delete(repo)
-        db.session.flush()
+                db.session.delete(repo)
+
+            db.session.flush()
+        else:
+            self.log_error("No matching repositories found")
```

Comment:
Nitpick:
`log_info` (log_warn) instead of `log_error`?

Reply:
Fixed.